### ISSUE TITLE
A `let` property cannot be reassigned after all fields have been initialized

### ISSUE THREAD

**Issue by mneumann** [savi-lang/savi#391]
```savi
:struct A
  :let field U8: 5

  :new (@field)
```

I'd assume that it is possible assign a value to a let field in a constructor and if not, the `:let` declarators default value is taken.

**Comment by jemc** [comment]
I agree this would be nice to have, but it will make the `completeness` analysis significantly more complicated, I think.

**Comment by mneumann** [comment]
I see. So we should better raise an error at compile time in these situation. IIRC, there is already an open issue for that.

**Comment by mneumann** [comment]
Sorry, mixing things up. Well, `:let` default values as of now are more or less constant as you cannot change them.

**Comment by jemc** [comment]
Right, I agree it would be more useful if default values for `:let` could be overridden by explicit values in a constructor, and right now, that is not possible.
